File: uc/computerScience2/spring2014/liveDemos/week03-pizza/pizza-06/topping.cpp
#include "topping.h"
#include <iostream>
using namespace std;

Topping::Topping() {
  // Nothing. The default constructor is not used.
}

Topping::Topping(string newName, bool isPremium) : name(newName) {
  premium = isPremium;
}

Topping::Topping(string newName) : name(newName) {
  premium = false;
}

Topping::~Topping() {
  // nothing to do.
}

string Topping::getName() {
  return name;
}

bool Topping::isPremium() {
  return premium;
}

File: uc/computerScience2/fall2013/liveDemos/java/pizza/Topping.java

public class Topping {
  private final String name;
  private final boolean premium;

  public Topping(String name, boolean isPremium) {
    this.name = name;
    this.premium = isPremium;
  }

  public Topping(String name) {
    // Call the other constructor!
    this(name, false);
  }

  private Topping() {
    // private, no use
    name = null;
    premium = false;
  }

  public String getName() {
    return name;
  }

  public boolean isPremium() {
    return premium;
  }
}